scripts/Inelastic/Direct/ReductionWrapper.py

```diff
@@ -368,12 +374,53 @@ def _run_pause(self,timeToWait=0):
         """
 
         if not self._debug_wait_for_files_operation is None:
-# it is callable and the idea is that it is callable
+# it is callable and the main point of this method is that it is callable
 #pylint: disable=E1102
             self._debug_wait_for_files_operation()
         else:
             Pause(timeToWait)
     #
+    def _check_access_granted(self,input_file):
+        """ Check if the access to the found nxs file is granted
+
+            Created to fix issue on ISIS archive, when file
+            is copied through the network for ~2min and become available
+            2 minutes after it has been found.
+        """
+
+        _,found_ext=os.path.splitext(input_file)
+        if found_ext != '.nxs': # problem solution for nxs files only. Others are seems ok
+            return
+        if not h5py_installed: # well this check is not available. Sad, but it available on
+            # all our working systems. Inform user about the problem
+            self.reducer.prop_man.log \
+            ('*** Can not verify if file is accessible. Install h5py to be able to check file access in waiting mode',\
+            'notice')
+            return
+        ic=0
+        #ok = os.access(input_file,os.R_OK) # does not work in this case
+        try:
+            f = h5py.File(input_file,'r')
```

Comment:
The pylint warning "invalid name" is disabled at the beginning of the file so this `f = ...` doesn't produce any warnings. But using more short variable names will make it harder in the future for anyone trying to remove naming convention issues.